Dockerfile

```diff
@@ -0,0 +1,31 @@
+# Use an official Python runtime as the parent image
+FROM python:3.10-slim
+
+# Set the working directory in the container
+WORKDIR /app
+
+# Copy the current directory contents into the container at /app
+COPY . /app
+
+# Install Poetry
+RUN pip install poetry
+
+# Install project dependencies using Poetry
+RUN poetry install --no-dev
+
+# Explicitly install uvicorn
+RUN pip install uvicorn
```

Comment:
all these should be replaced by 
`pip install poetry`
`poetry install`
(like in Makefile)

Reply:
this step I couldn't do it because I had to install all the required libraries in the code explicitly due peotry did not adequately meet the needs of the code. And after completing the process of building the image, errors appear when running the Docker image in which the code requests to download the libraries from which the code calls functions.

So I had to keep it as it is. If you have any other comments on this step I would be happy to hear them.

Thanks